core/commonMain/src/kotlinx/io/buffer/Buffer.kt

```diff
@@ -59,3 +59,8 @@ public expect class Buffer {
         public val EMPTY: Buffer
     }
 }
+
+/**
+ * Wrap [array] into [Buffer] from [startIndex] to [endIndex].
+ */
+expect fun bufferOf(array: ByteArray, startIndex: Int = 0, endIndex: Int = array.size): Buffer
```

Comment:
Do we need this function to be `public`? Let's postpone adding new syntactic sugar while we are still at the initial phase

Reply:
I used it in samples a lot. Let's mark it with experimental `@InternalAPI`.

Reply:
It is effectively unused and untested.
Let's introduce it along with samples. Add a shortcut is much easier than remember that one is not used and can be safely deleted.